src/name/bizna/ocarmsim/FakeMachine.java

```diff
@@ -189,15 +201,14 @@ public Architecture architecture() {
 
 	@Override
 	public Map<String, String> components() {
-		Map<String, String> ret = new HashMap<String, String>();
+		Map<String, String> ret = new HashMap<>();
```

Comment:
My knowledge of Java generics is pretty poor, but doesn't this create a warning where there wasn't one before?